src/OpenTelemetry.Api/Trace/EventNew.cs

```diff
@@ -0,0 +1,106 @@
+﻿// <copyright file="EventNew.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Diagnostics;
+using OpenTelemetry.Api.Utils;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// A text annotation associated with a collection of attributes.
+    /// </summary>
+    public sealed class EventNew
+    {
+        internal ActivityEvent ActivityEvent;
+        private static readonly ReadOnlyDictionary<string, object> EmptyAttributes =
```

Comment:
EmptyAttributes doesn't seem to be used for anything? Same comment for `LinkNew`.